crates/orderbook/src/account_balances.rs

```diff
@@ -215,6 +227,32 @@ impl BalanceFetching for Web3BalanceFetcher {
         };
         Ok(success)
     }
+
+    async fn has_sufficient_balance(&self, token: H160, from: H160, amount: U256) -> Result<bool> {
+        let instance = ERC20::at(&self.web3, token);
+        let balance = instance.balance_of(from).call().await?;
+        Ok(balance >= amount)
+    }
+
+    async fn has_sufficient_allowance(
+        &self,
+        token: H160,
+        from: H160,
+        amount: U256,
+        source: SellTokenSource,
+    ) -> Result<bool> {
+        let success = match source {
+            SellTokenSource::Erc20 => {
+                let instance = ERC20::at(&self.web3, token);
+                instance.allowance(from, self.vault_relayer).call().await? >= amount
+            }
+            SellTokenSource::External => {
+                self.can_manage_user_balance_call(token, from, amount).await
```

Comment:
One slight worry is that this has an implicit requirement that `balance_fetcher.has_sufficient_balance()` was previously checked, otherwise `has_sufficient_allowance` would generate false-positives when the user has a sufficient ERC20 allowance on the Vault and set relayer approval but has not more ERC20 token balance.